### PR TITLE
fix: add txHex for eth CCR tx

### PR DESCRIPTION
**Pull request by kamleshmugdiya**
Ticket: COIN-119

### CODE REVIEW

File: modules/sdk-coin-eth/src/eth.ts [hunk 1]
```diff
@@ -429,6 +429,9 @@ export class Eth extends AbstractEthLikeNewCoins {
    * @returns {Promise<SignedTransaction>}
    */
   async signTransaction(params: SignTransactionOptions): Promise<SignedTransaction> {
+    if (params.isEvmBasedCrossChainRecovery) {
+      return super.signTransaction(params);
+    }
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-eth/src/eth.ts:434]: please add a test to ensure, its calling the right function

Comment (margueriteblair) [reply modules/sdk-coin-eth/src/eth.ts:434]: +1

Comment (kamleshmugdiya) [reply modules/sdk-coin-eth/src/eth.ts:434]: added a test

File: modules/sdk-coin-eth/test/unit/eth.ts [hunk 2]
```diff
@@ -95,6 +103,65 @@ describe('ETH:', function () {
...
+          txHex: tx.toBroadcastFormat(),
+        },
+        prv: userKeychain.prv,
+        isEvmBasedCrossChainRecovery: true,
+      } as any);
+
+      (halfSignedTransaction as any).halfSigned.eip1559.maxPriorityFeePerGas.should.equal(10);
+      console.log(halfSignedTransaction);
+      assert((halfSignedTransaction as any).halfSigned.txHex !== undefined);
+      (halfSignedTransaction as any).halfSigned.eip1559.maxFeePerGas.should.equal(10);
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-eth/test/unit/eth.ts:161]:
```suggestion
      assert((halfSignedTransaction as any).halfSigned.txHex);
      assert.strictEqual((halfSignedTransaction as any).halfSigned.eip1559.maxFeePerGas,10);
```

File: modules/sdk-coin-eth/test/unit/eth.ts [hunk 3]
```diff
@@ -51,12 +56,15 @@ describe('ETH:', function () {
       bitgo.safeRegister(name, coinConstructor);
     });
     bitgo.safeRegister('teth', Teth.createInstance);
+    bitgo.safeRegister('hteth', Teth.createInstance);
```
[the remark this answers is not in the document]
Comment (kamleshmugdiya) [reply modules/sdk-coin-eth/test/unit/eth.ts]: thanks, fixed it.

**Review by mullapudipruthvik — APPROVED** [review]

**Pull request merged**